lsp-mode.el

```diff
@@ -8091,6 +8091,12 @@ such."
       (lsp--warn "No LSP server for %s(check *lsp-log*)." major-mode)
       nil)))
 
+(defun lsp--any-inconsistency? ()
+  "Check for any inconsistency on LSP package"
+  (and lsp-use-plists
+       (or (not (fboundp 'lsp-make-range))
```

Comment:
Is this check correct? It just check for function that will be defined when `lsp-mode` is required (with `lsp-protocol`.)
I think the check should be something along
```
(xor lsp-use-plists
     (equal (symbol-function 'lsp-merge) 'append))
```

Reply:
It checks if `lsp-make-*` that is a function that only works when lsp-mode is re-compiled correctly, right?
I don't know for sure the difference between using `(fboundp 'lsp-make-range)` and `(symbol-function 'lsp-make-range)` for checking if non-nil, I think both works for what we need here, am I wrong?

Reply:
> It checks if `lsp-make-*` that is a function that only works when lsp-mode is re-compiled correctly, right?

Not really, the `lsp-make-*` is defined regardless of state of `lsp-use-plists`.
So for example, you have been in recent `lsp-mode` with `lsp-use-plists` is `nil`, then there's update where `lsp-use-plists` become `t`, `(fboundp 'lsp-make*)` will still return `t` even if you don't recompile `lsp-mode` correctly.

On other hand, `lsp-merge`, which is aliased to `ht-merge` or `append` depends on `lsp-use-plists` will just return `ht-merge` if you don't recompile properly when switch to `lsp-use-plists`

Reply:
I see, It make sense, thanks for the explanation, I'll test it locally then change it!

Reply:
Fixed it!